app/assets/javascripts/thredded.es6

```diff
@@ -1,18 +1,59 @@
+//= require babel/polyfill
 //= require jquery
 //= require jquery_ujs
 //= require_tree ./thredded/vendor
 //= require_tree ./thredded
 
-class Thredded {
-  constructor() {
-    this.topicForm = new ThreddedTopicForm();
-    this.privateTopicForm = new ThreddedPrivateTopicForm();
-    this.postForm = new ThreddedPostForm();
-    this.kbShortcuts = new ThreddedKeyboardShortcuts();
-    this.timestamps = new ThreddedTimeStamps();
-    this.searchBar = new ThreddedSearchBar();
-    this.currentlyOnline = new ThreddedCurrentlyOnline();
+(function() {
+
+  const pageComponents = [
+    ['thredded-edit-post', 'postForm'],
+    ['thredded-new-private-topic', 'postForm', 'privateTopicForm'],
+    ['thredded-new-topic', 'postForm', 'topicForm'],
+    ['thredded-posts', 'postForm'],
+    ['thredded-private-topics-index', 'privateTopicForm'],
+    ['thredded-theme', 'topicForm', 'privateTopicForm', 'postForm', 'searchBar'],
+    ['thredded-topics-index', 'topicForm']
+  ];
```

Comment:
I like this a lot! Awesome approach :). I wonder, though, if it could be made a little more clear.

```javascript
const pageComponents = {
  { pageId: 'thredded-edit-post',            components: ['postForm'] },
  { pageId: 'thredded-new-private-topic',    components: ['postForm', 'privateTopicForm'] },
  { pageId: 'thredded-new-topic',            components: ['postForm', 'topicForm'] },
  { pageId: 'thredded-posts',                components: ['postForm'] },
  { pageId: 'thredded-private-topics-index', components: ['privateTopicForm'] },
  { pageId: 'thredded-theme',                components: ['topicForm', 'privateTopicForm', 'postForm', 'searchBar'] },
  { pageId: 'thredded-topics-index',         components: ['topicForm'] },
}
```

This could be splitting hairs a little bit but I did have to sit and think for a second before I realized that there was a fundamental difference between items at index 0 vs 1+ ... so it's coming from a place of communicating intent, not pedantry ;)